src/wp-includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,884 @@
+	 *
+	 * Keyed on the dependency's slug.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	protected static $dependency_filepaths = array();
+
+	/**
+	 * Initializes by fetching plugin header and plugin API data,
+	 * and deactivating dependents with unmet dependencies.
+	 *
+	 * @since 6.4.0
+	 */
+	public static function initialize() {
+		self::read_dependencies_from_plugin_headers();
+		self::get_dependency_api_data();
+		self::deactivate_dependents_with_unmet_dependencies();
+	}
+
+	/**
+	 * Determines whether the plugin has plugins that depend on it.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $plugin_file The plugin's filepath, relative to the plugins directory.
+	 * @return bool Whether the plugin has plugins that depend on it.
+	 */
+	public static function has_dependents( $plugin_file ) {
+		$dependency_paths = self::get_dependency_filepaths();
+		return in_array( $plugin_file, $dependency_paths, true );
+	}
+
+	/**
+	 * Determines whether the plugin has plugin dependencies.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $plugin_file The plugin's filepath, relative to the plugins directory.
+	 * @return bool Whether a plugin has plugin dependencies.
+	 */
+	public static function has_dependencies( $plugin_file ) {
+		return isset( self::$dependencies[ $plugin_file ] );
+	}
+
+	/**
+	 * Determines whether the plugin has active dependents.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $plugin_file The plugin's filepath, relative to the plugins directory.
+	 * @return bool Whether the plugin has active dependents.
+	 */
+	public static function has_active_dependents( $plugin_file ) {
+		$dependents = self::get_dependents( dirname( $plugin_file ) );
+
+		foreach ( $dependents as $dependent ) {
+			if ( is_plugin_active( $dependent ) ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Determines whether the plugin has unmet dependencies.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $plugin_file The plugin's filepath, relative to the plugins directory.
+	 * @return bool Whether the plugin has unmet dependencies.
+	 */
+	public static function has_unmet_dependencies( $plugin_file ) {
+		if ( ! isset( self::$dependencies[ $plugin_file ] ) ) {
+			return false;
+		}
+
+		foreach ( self::$dependencies[ $plugin_file ] as $dependency ) {
+			$dependency_filepath = self::get_dependency_filepath( $dependency );
+
+			if ( false === $dependency_filepath || is_plugin_inactive( $dependency_filepath ) ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Gets the names of plugins that require the plugin.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array $plugin_data Array of plugin data.
+	 * @return array An array of dependent names.
+	 */
+	public static function get_dependent_names( $plugin_file ) {
+		$dependent_names = array();
+
+		if ( empty( self::$plugins ) ) {
+			self::$plugins = get_plugins();
+		}
+
+		$slug = str_contains( $plugin_file, '/' ) ? basename( dirname( $plugin_file ) ) : $plugin_file;
```

Comment:
```suggestion
		$slug = str_contains( $plugin_file, '/' ) ? dirname( $plugin_file ) : $plugin_file;
```
No need to `basename()`. Everything is stated to be relative to the plugins directory.